src/components/Layout/components/Treeview/index.js

```diff
@@ -54,11 +55,20 @@ const treelist = [
 ];
 
 const Treeview = () => {
+  const { getProductsByCategory } = useContext(ProductContext);
+  const fetchByCategory = (category) => {
+    getProductsByCategory(category);
+  };
+
   const renderTree = (categories) => {
     let renderList = [];
     for (let category of categories) {
       renderList.push(
-        <TreeItem nodeId={category.title} label={category.title}>
+        <TreeItem
+          nodeId={category.title}
+          label={category.title}
+          onClick={() => fetchByCategory(category.title)}
```

Comment:
```suggestion
          onClick={() => getProductsByCategory(category.title)}
```
tại sao k phải như này luôn nhờ